main/modes/picross/mode_picross.c

```diff
@@ -0,0 +1,1167 @@
+            ((i * s) + s + p->leftPad) >> 1,
+            ((p->puzzle->height * s) + s + p->topPad) >> 1,
+            c115,0);
+            continue;
+        }
+        if(i%5 == 0)//draw thicker line every 5.
+        {
+            box_t box =
+            {
+                .x0 = (i * s) + s + p->leftPad - w/2,
+                .y0 = s + p->topPad,
+                .x1 = (i * s) + s + w/2 + p->leftPad,//3 is width
+                .y1 = (p->puzzle->height * s) + s + p->topPad,
+            };  
+            drawBox(d,box,c441,true,1);
+            continue;
+        }
+        //this could be less confusing.
+        plotLine(d,
+            ((i * s) + s + p->leftPad) >> 1,
+            (s + p->topPad) >> 1,
+            ((i * s) + s + p->leftPad) >> 1,
+            ((p->puzzle->height * s) + s + p->topPad) >> 1,
+            c111,0);
+        
+    }
+
+     //draw a horizontal line every 5 units.
+     //todo: also do the full grid.
+    for(int i=0;i<=p->puzzle->height;i++)
+    {
+        uint8_t s = p->drawScale;
+        if(i == 0 || i == p->puzzle->width)
+        {
+            //draw border
+            plotLine(d,
+            (s + p->leftPad) >> 1,
+            ((i * s) + s + p->topPad) >> 1,
+            ((p->puzzle->width * s) + s + p->leftPad) >> 1,
+            ((i * s) + s + p->topPad) >> 1,
+            c115,0);
+            continue;
+        }
+        if(i%5 == 0)
+        {
+            box_t box =
+            {
+                .x0 = s + p->leftPad,
+                .y0 = (i * s) + s + p->topPad - w/2,
+                .x1 = (p->puzzle->width * s) + s + p->leftPad,
+                .y1 = (i * s) + s + p->topPad + w/2,
+            };  
+            drawBox(d,box,c441,true,1);
+            continue;
+        }
+        //this should be less confusing.
+        plotLine(d,
+            (s + p->leftPad) >> 1,
+            ((i * s) + s + p->topPad) >> 1,
+            ((p->puzzle->width * s) + s + p->leftPad) >> 1,
+            ((i * s) + s + p->topPad) >> 1,
+            c111,0);
+     
+    }
+
+    //Draw hints
+    for(int i=0;i<10;i++)
+    {
+        drawHint(d,font,p->puzzle->rowHints[i]);
+        drawHint(d,font,p->puzzle->colHints[i]);
+    }
+}
+
+void drawHint(display_t* d,font_t* font, picrossHint_t hint)
+{
+    int8_t vHintShift = 2;
+    uint8_t h;
+    uint8_t g = p->clueGap;//
+    //todo: handle 10 or double-digit input 
+    if(hint.isRow){
+        int j = 0;
+        for(int i = 0;i<5;i++)
+        {
+            h = hint.hints[4-i];
+            box_t hintbox = boxFromCoord(-j-1,hint.index);
+            hintbox.x0 = hintbox.x0 - (g * (j));
+            hintbox.x1 = hintbox.x1 - (g * (j));
+            //we have to flip the hints around. 
+            if(h == 0){
+                //dont increase j.
+                
+            }else if(h < 10){
+                
+                //drawBox(d,hintbox,c111,false,1);//for debugging. we will draw nothing when proper.
+
+                char letter[1];
```

Comment:
I think this needs to be bigger, one char for `h` and one for a null terminator.

Reply:
If you want to be a little safer, you can always use `snprintf()`

Reply:
You could also do `char c = ('0' + h);`, but that only works for one digit numbers